src/main/kotlin/com/projectcitybuild/spigot/modules/chat/listeners/ChatListener.kt

```diff
@@ -29,30 +31,32 @@ class ChatListener : Listenable<AsyncPlayerChatEvent> {
         val lpUser = permissions.userManager.getUser(event.player.uniqueId)
                 ?: throw Exception("Could not load user from LuckPerms")
 
-        val groupNodes = lpUser.ownNodes.stream()
-                .filter(Node::isGroupNode)
-                .map { node -> permissions.groupManager.getGroup(node.groupName) }
+        val groupNodes = lpUser.nodes.stream()
+                .filter(NodeType.INHERITANCE::matches)
+                .map(NodeType.INHERITANCE::cast)
                 .collect(Collectors.toSet())
 
-        val prefixes = lpUser.ownNodes.stream()
-                .filter(Node::isPrefix)
-                .map { node -> node.prefix.value }
+        val prefixes = lpUser.nodes.stream()
+                .filter(NodeType.PREFIX::matches)
+                .map(NodeType.INHERITANCE::cast)
+                .map {  node -> node.value }
                 .collect(Collectors.toSet())
                 .joinToString(separator = "")
 
-        val suffixes = lpUser.ownNodes.stream()
-                .filter(Node::isSuffix)
-                .map { node -> node.suffix.value }
+        val suffixes = lpUser.nodes.stream()
+                .filter(NodeType.SUFFIX::matches)
+                .map(NodeType.INHERITANCE::cast)
+                .map { node -> node.value }
                 .collect(Collectors.toSet())
                 .joinToString(separator = "")
 
         val groups = mutableListOf<String>()
         groupNodes.forEach { group ->
             val group = group ?: return
-            val displayName = group.displayName ?: group.name
+            val displayName = group.groupName
```

Comment:
I wasn't able to figure out the distinction of display names and names in the new API, so for now this just uses `groupName`

Reply:
Looks like we might have to get the group and see if it has any `DisplayNameNode`s?

https://github.com/lucko/LuckPerms/wiki/Developer-API:-Usage#the-basics-of-node

Reply:
It turns out you can just fetch the group from `groupManager` and directly access the `displayName` property 👍